app/scripts/views/form.js

```diff
@@ -199,6 +199,10 @@ define(function (require, exports, module) {
 
     _submitForm: notifyDelayedRequest(showButtonProgressIndicator(function () {
       var self = this;
+      // sets all password fields to type=password
+      self.$el.find('.password').each(function (i, el) {
```

Comment:
Adding it there will need an event handler for submit as well, which is related to the form, and not the password field. I think it would be better to keep it here.
@shane-tomlinson